### PR TITLE
Improve schema instructions

### PR DESCRIPTION
**Pull request by glerb**
Added some emphasis to key parts of the process, and some guidance about how to frontload a request for a new log source type.

### CODE REVIEW

File: docs/gitbook/development/writing-parsers.md [hunk 1]
```diff
@@ -75,11 +84,9 @@ result: |
   }
 ```
 
-Note that Panther adds some fields to the output event. We call these fields ['panther fields'](../data-analytics/panther-fields.md) and they are prefixed with `p_` to avoid name collisions.
+Note the additional fields beginning with `p_` appended to the `result`. We call these fields ['panther fields'](../data-analytics/panther-fields.md) and they are prefixed with `p_` to avoid name collisions. To cause these fields to appear in processed logs, a special tag called a `scanner` must be added to any raw event fields that you wish to appear in a given panther field (see "Indicator Strings", below).
```
Comment (papanikge) [inline docs/gitbook/development/writing-parsers.md:87]: That's not exactly correct. Some special fields are added whatsoever. example: `p_event_time`

File: docs/gitbook/development/writing-parsers.md [hunk 2]
```diff
@@ -37,7 +46,7 @@ Our goal here is to parse a JSON string of a `Foo.Event` and produce a Panther l
 
 Panther provides the `logtesting` package with helpers for testing log parsers. This module allows to run tests declared in YAML files using the `logtesting.RunTestsFromYAML` function. These tests verify that the parser produces the expected log event(s) for a log entry. This method ensures we haven't missed any fields or indicator values throughout the process.
 
-We write a test for `Foo.Event` in a YAML file at `foologs/testdata/foologs_tests.yml`
+We write a test for `Foo.Event` in a YAML file at `foologs/testdata/foologs_tests.yml`. The `input` is the raw log from the log source, and the `result` is the log after it has been processed by the log processor:
```
Comment (papanikge) [inline docs/gitbook/development/writing-parsers.md:49]: maybe a better wording:
```
[...] processed by panther
```

File: docs/gitbook/development/writing-parsers.md [hunk 3]
```diff
@@ -125,17 +132,25 @@ type User struct {
...
+Note: The `json:"omitempty` tag is not the same as the [validator tag](https://pkg.go.dev/gopkg.in/go-playground/validator.v9#hdr-Omit_Empty) `validate:"omitempty".
 
 #### String values
 
 > `pantherlog.String`
 
 #### Indicator strings
 
-String values can be tagged with a `panther:"SCANNER"` to define [indicator fields](../data-analytics/panther-fields.md#indicator-fields). Note that a scanner can produce multiple indicator fields from a single value, or a different indicator field based on the value. Panther defines the following scanners
+String values can be tagged with a `panther:"<scanner>"` to define [indicator fields](../data-analytics/panther-fields.md#indicator-fields). Note that a `scanner` can produce multiple indicator fields from a single value, or a different indicator field based on the value. Panther defines the following scanners:
```
Comment (papanikge) [inline docs/gitbook/development/writing-parsers.md:153]: do we still call them scanners? 🤔  aren't we use indicators more often nowadays? not sure

File: docs/gitbook/development/writing-parsers.md [hunk 4]
```diff
@@ -125,17 +132,25 @@ type User struct {
...
+The `RawMessage` type is for a JSON blob for which you don't wish to create sub-`struct` objects. Note that in order to search for nested attributes and values in your logs using [JSONPath](https://goessner.net/articles/JsonPath/)-like syntax in Panther Data Explorer, you must define sub-`struct` types in your log schema file. You can still search `RawMessage` fields, but you will have to use the `json_extract` [function in Athena](https://prestodb.io/docs/0.217/functions/json.html#json_extract) or the `GET_PATH` [function in Snowflake](https://docs.snowflake.com/en/sql-reference/functions/get_path.html).
+
+Panther uses [struct tags](https://www.digitalocean.com/community/tutorials/how-to-use-struct-tags-in-go) to specify field attributes. All fields _must_ have some `description` to document the contents of this field. These documentation strings will be used to generate user documentation from this code. The `json` tag _must_ use the exact field name used in the raw log. Panther automatically adds the `omitempty` option to the `json:` tag for all fields.
```
Comment (papanikge) [inline docs/gitbook/development/writing-parsers.md:141]: well, it doesn't "adds the `omitempty` option" per-se, it just follows that behavior

File: docs/gitbook/development/writing-parsers.md [hunk 5]
```diff
@@ -125,17 +132,25 @@ type User struct {
...
+
+Panther relies on the `validate:` tag to distinguish different log types. The [validator package](https://pkg.go.dev/gopkg.in/go-playground/validator.v9) defines the syntax for this tag. Panther specifically relies on the `validate:"required"` tag, which tells Panther which fields must be present in order to classify a log as a specific type. _The `validate:"required"` tag should be applied to the fewest fields absolutely necessary to identify a given log type_ to minimize processing overhead in Panther. Common fields such as timestamps generally should not be tagged as `"required"`. Additionally, before tagging a field as `"required"`, review a sample of log events from your source to verify that the field is present in _every_ log. _If a `"required"` field is not present in a log event, it will not be correctly identified. It may be dropped or incorrectly classified as a different log type_.
```
Comment (papanikge) [inline docs/gitbook/development/writing-parsers.md:143]: As far as I know, that's not exactly true. We classify the events with the parser, by trying each one, one at a time, until one succeeds. We don't rely "with code" on the validation tag, it's just that the mandatory fields are the only ones that can we can rely on both being there and for differentiating with other logtypes.

**Review by papanikge — COMMENTED** [review]
Nice

File: docs/gitbook/development/writing-parsers.md (hunk 3 above)
Comment (glerb) [reply docs/gitbook/development/writing-parsers.md:153]: I don't know.

File: docs/gitbook/development/writing-parsers.md (hunk 4 above)
Comment (glerb) [reply docs/gitbook/development/writing-parsers.md:141]: That was the text in the original version. I didn't change it.

File: docs/gitbook/development/writing-parsers.md (hunk 5 above)
Comment (glerb) [reply docs/gitbook/development/writing-parsers.md:143]: Which part isn't true?

File: docs/gitbook/development/writing-parsers.md (hunk 1 above)
Comment (glerb) [reply docs/gitbook/development/writing-parsers.md:87]: Do we have a list of the ones that are automatically added?

**Pull request closed without merging**